### PR TITLE
Added Pubsub to MongoDB flex template

### PR DESCRIPTION
**Pull request by pritukam**

### CODE REVIEW

File: v2/pubsub-to-mongodb/src/main/java/com/google/cloud/teleport/v2/templates/PubSubToMongoDB.java [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+  /**
+   * DoFn that will parse the given string elements as Bson Documents.
+   */
+  public static class ParseAsDocumentsFn extends DoFn<String, Document> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ParseAsDocumentsFn.class);
+
+    @ProcessElement
+    public void processElement(ProcessContext context) {
+      context.output(Document.parse(context.element()));
```
Comment (sabhyankar) [inline v2/pubsub-to-mongodb/src/main/java/com/google/cloud/teleport/v2/templates/PubSubToMongoDB.java]: Will this parsing always be successful? If not, we should put this in a try-catch block and use separate tags to catch these parsing failures (with metric-counters).

File: v2/pubsub-to-mongodb/src/test/com/google/cloud/teleport/v2/templates/PubSubToMongoDBTest.java [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
```
Comment (sabhyankar) [inline v2/pubsub-to-mongodb/src/test/com/google/cloud/teleport/v2/templates/PubSubToMongoDBTest.java]: Missing javadoc.

File: v2/pubsub-to-mongodb/src/test/com/google/cloud/teleport/v2/templates/PubSubToMongoDBTest.java [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.teleport.v2.templates;
+
+import com.google.cloud.teleport.v2.values.FailsafeElement;
```
Comment (sabhyankar) [inline v2/pubsub-to-mongodb/src/test/com/google/cloud/teleport/v2/templates/PubSubToMongoDBTest.java]: Can you reformat/reorder the imports?

File: v2/pubsub-to-mongodb/src/main/java/com/google/cloud/teleport/v2/templates/PubSubToMongoDB.java (hunk 1 above)
Comment (pritukam) [reply v2/pubsub-to-mongodb/src/main/java/com/google/cloud/teleport/v2/templates/PubSubToMongoDB.java]: The function will always get a valid json. Valid json will always get parsed as Bson documents

Comment (sabhyankar) [reply v2/pubsub-to-mongodb/src/main/java/com/google/cloud/teleport/v2/templates/PubSubToMongoDB.java]: Yes in the context of this pipeline. However, what happens when I attempt to use the public static class and pass a String (which is not a valid json)?

**Pull request merged**